lib/matplotlib/_type1font.py

```diff
@@ -35,10 +37,64 @@
 
 from matplotlib.cbook import _format_approx
 from . import _api
+if T.TYPE_CHECKING:
+    from collections.abc import Iterable
```

Comment:
We tend not to have inline type hints (and I personally very much don't like them), but there are a few exceptions (e.g. _mathtext.py) so I guess it's up to you whether you want to leave them in.

Reply:
Right – I saw that there are some files with type hints, and figured that the project might be in the process of adding them. I've found type hints pretty useful at work, but of course we should have a consistent style in the project. Has this been discussed in the past on the dev list or somewhere?

Reply:
I'm probably among the most vocal opponents, but I'll defer to @QuLogic or @ksunden on this aspect.

Reply:
and @tacaswell responded

> It is worth re-opening the discussion of in-line type hints
the world seems to have stabilized and in other projects I have had type hints catch actual bugs before I ran the code (but I have also had to spend 15 minutes trying to sort out how to placate the type checker for code that clearly works a couple of times)


The type hints were useful for me while working on this, since VS Code pointed out some obvious mistakes in real time. But I agree that it is not ideal to leave the file only partially annotated, since it does not pass a full type check in its current state.

I could remove the extra type hints from this PR and make a separate PR to add hints to the whole file?